test/xbgpu/test_engine.py

```diff
@@ -1150,14 +1156,10 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
 
             await xbengine.stop()
 
-        assert (
-            prom_diff.get_sample_diff("output_x_visibilities_total", {"stream": "bcp1"})
-            == n_channels_per_substream * n_baselines
-        )
-        assert (
-            prom_diff.get_sample_diff("output_x_clipped_visibilities_total", {"stream": "bcp1"})
-            == n_channels_per_substream * n_baselines
-        )
+        n_vis = n_channels_per_substream * n_baselines
+        assert prom_diff.get_sample_diff("output_x_visibilities_total", {"stream": "bcp1"}) == n_vis
+        assert prom_diff.get_sample_diff("output_x_clipped_visibilities_total", {"stream": "bcp1"}) == n_vis
+        assert xbengine.sensors["bcp1.xeng-clip-cnt"].value == n_vis
```

Comment:
Ha, nice one. I do feel bad for asking you to loop this over each `corrprod_output`. Not a priority, I'll create an `if-bored` ticket.

Reply:
I'm just being consistent with what's already been done for Prometheus.